clang/lib/Sema/SemaInit.cpp

```diff
@@ -10431,7 +10437,7 @@ static void DiagnoseNarrowingInInitList(Sema &S,
                                         : diag::warn_init_list_type_narrowing)
         << PostInit->getSourceRange()
         << PreNarrowingType.getLocalUnqualifiedType()
-        << EntityType.getLocalUnqualifiedType();
+        << EntityType.getLocalUnqualifiedType().getNonReferenceType();
```

Comment:
Since this is supposed to remove local qualifiers, should the reference be removed first?  What happens with a `const int &` as is?

ALSO, does the pre-narrowing-type also need to have references removed?

Reply:
> Since this is supposed to remove local qualifiers, should the reference be removed first? What happens with a const int & as is?

Perhaps makes sense. I only removed reference since it is already removed from the pre-narrowing-type and I was a bit inspired by gcc messages https://godbolt.org/z/Tcv17Wxhc . I could leave reference type it as is, then the messages will look like:

```

struct A { A(const unsigned &x) {} };

int foo(const int &aba) {
    A a { -1 }; // constant expression evaluates to -1 which cannot be narrowed to type 'const unsigned int&'
    A b { aba }; // non-constant-expression cannot be narrowed from type 'int' to 'const unsigned int' in initializer list
    return 0;
}
```

Reply:
Why does leaving the reference type alone (which seems to be removed 'last' cause the const to stick around, and not the reference?

Reply:
My guess is `getLocalUnqualifiedType` doesn't strip cv qualifiers from reference's pointee type, so we first should remove reference and then qualifiers to see plain `int` in the diagnostic.

Reply:
I think we're talking past eachother then.  I was suggesting to remove the reference first, which would have us match GCC's behavior, right?  I think that is better than leaving it 'as is', but I was confused as to what is going on with the change on 10440 (removing the reference AFTER the qualifiers instead of before);